qemu_v8.mk

```diff
@@ -41,15 +44,54 @@ QEMU_PATH		?= $(ROOT)/qemu
 QEMU_BUILD		?= $(QEMU_PATH)/build
 SOC_TERM_PATH		?= $(ROOT)/soc_term
 MODULE_OUTPUT		?= $(ROOT)/out/kernel_modules
-
+UBOOT_PATH		?= $(ROOT)/u-boot
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+MKIMAGE_PATH		?= $(UBOOT_PATH)/tools
+
+ROOTFS_GZ		?= $(BINARIES_PATH)/rootfs.cpio.gz
+ROOTFS_UGZ		?= $(BINARIES_PATH)/rootfs.cpio.uboot
+
+KERNEL_IMAGE		?= $(LINUX_PATH)/arch/arm64/boot/Image
+KERNEL_IMAGEGZ		?= $(LINUX_PATH)/arch/arm64/boot/Image.gz
+KERNEL_UIMAGE		?= $(BINARIES_PATH)/uImage
+
+# Load and entry addresses
+KERNEL_ENTRY		?= 0x40400000
+KERNEL_LOADADDR		?= 0x40400000
+ROOTFS_ENTRY		?= 0x44000000
+ROOTFS_LOADADDR		?= 0x44000000
+
+ifeq ($(UBOOT),y)
+BL33_BIN		?= $(UBOOT_BIN)
+BL33_DEPS		?= u-boot
+else
+BL33_BIN		?= $(EDK2_BIN)
+BL33_DEPS		?= edk2
+endif
 
 ################################################################################
 # Targets
 ################################################################################
-all: arm-tf buildroot edk2 linux optee-os qemu soc-term
-clean: arm-tf-clean buildroot-clean edk2-clean linux-clean optee-os-clean \
+TARGET_DEPS := arm-tf buildroot linux optee-os qemu soc-term
+TARGET_CLEAN := arm-tf-clean buildroot-clean linux-clean optee-os-clean \
 	qemu-clean soc-term-clean check-clean
 
+TARGET_DEPS 		+= $(BL33_DEPS)
+
+ifeq ($(UBOOT),y)
+TARGET_DEPS		+= $(KERNEL_UIMAGE) $(ROOTFS_UGZ)
+TARGET_CLEAN		+= u-boot-clean
+else
+TARGET_CLEAN		+= edk2-clean
+endif
+
+all: $(TARGET_DEPS)
```

Comment:
These dependencies without actions can be repeated as many times as needed so there's generally not a need to do this trick with a variable. There's not much harm in doing like this either so keep this if you prefer that.

Reply:
I just found it a little more readable hence did it this way.